react/components/results.js

```diff
@@ -7,21 +7,51 @@ import { graphSearchIPAddress } from '../constants'
 import './results.scss'
 
 const ResultsComponent = React.createClass({
+  componentDidMount() {
+    window.addEventListener('scroll', this.handleScroll)
+  },
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleScroll)
+  },
+
   handleLoadMore() {
     this.props.getPage(this.props.query, this.props.results.length)
   },
 
+  isElementInViewport(el) {
+    const rect = el.getBoundingClientRect()
+
+    return (
+      rect.top >= 0 &&
+      rect.left >= 0 &&
+      Math.floor(rect.bottom) <=
+        (window.innerHeight ||
+          document.documentElement.clientHeight) /*or $(window).height() */ &&
+      Math.floor(rect.right) <=
+        (window.innerWidth ||
+          document.documentElement.clientWidth) /*or $(window).width() */
+    )
+  },
```

Comment:
here is where I define `isElementInViewport` 

but yes, the pattern is really useful!